File: lc_2026_04_24b.cpp
// Furthest Dist From Origin
class Solution {
public:
    int furthestDistanceFromOrigin(string moves) {

    int left = 0;
    int neut = 0;
    int right = 0;

    for( int i = 0; i< moves.length(); i++){
        if (moves[i] == 'L'){
            left++;
        }
        else if (moves[i] == 'R'){
            right++;
        }
        else if (moves[i] == '_'){
            neut++;
        }
    }

    if (left > right){
        return (left + neut - right);
    }
    else{
        return (right + neut - left);
    } 
    }
};

File: lc_2026_04_24b.rs
//Furthest Dist From Origin
impl Solution {
    pub fn furthest_distance_from_origin(moves: String) -> i32 {

        let mut left: i32 = 0;
        let mut right: i32 = 0;
        let mut neut: i32 = 0;

        for ch in moves.chars(){
            if ch == 'R'{
                right += 1;
            }
            else if ch == 'L'{
                left += 1;
            }
            else if ch == '_'{
                neut += 1;
            }
        }

        if left > right{
            left + neut - right
        }
        else{
            right + neut - left
        }
    }
}
